test/test-tcp-rst.c

```diff
@@ -47,9 +47,9 @@ static void read_cb(uv_stream_t* t, ssize_t nread, const uv_buf_t* buf) {
   ASSERT_PTR_EQ((uv_tcp_t*) t, &tcp);
   ASSERT_EQ(nread, UV_ECONNRESET);
 
-  int fd;
+  uv_os_fd_t fd;
   ASSERT_OK(uv_fileno((uv_handle_t*) t, &fd));
-  uv_handle_type type = uv_guess_handle(fd);
+  uv_handle_type type = uv_guess_handle(uv_open_osfhandle(fd));
```

Comment:
Note that you need to call `DuplicateHandle` first on `fd` before it is valid to call `uv_open_osfhandle` on it and then call `_close` afterwards

Reply:
Right. I'm tempted to remove the check altogether, since it's not related to the test at all...